src/index.js

```diff
@@ -99,15 +108,41 @@ function APP() {
   // …some time passes
   // (3) This gets called, and we write back the current query string to the browser's location using setSearchParams
   useEffect(() => {
-    // For now, encode only the selected search facets and start page in the searchParams
-    const searchFields = encode(JSON.stringify(store.getState()['query']['searchFields']));
-    const start = encode(JSON.stringify(store.getState()['query']['start'] / store.getState()['query']['rows']));
-    const sortFields = encode(JSON.stringify(store.getState()['query']['sortFields']));
-    const searchParamsDict = { searchFields, start, sortFields };
+    const query = store.getState()['query'];
+    // Only convert the fields with value as the state rather than all fields
+    const searchParamsDict = {};
+    if (checkAllValue(query['searchFields'])) {
+      const searchFields = encode(JSON.stringify(getAllValueField(query['searchFields'])));
+      searchParamsDict['searchFields'] = searchFields;
+    }
+
+    if (checkAllValue(query['sortFields'])) {
+      const sortFields = encode(JSON.stringify(getAllValueField(query['sortFields'])));
+      searchParamsDict['sortFields'] = sortFields;
+    }
+
+    if (query['start'] !== 0) {
+      const start = encode(JSON.stringify(query['start'] / query['rows']));
+      searchParamsDict['start'] = start;
+    }
+
+    if (JSON.stringify(query['view']) !== JSON.stringify(initialCamera)) {
+      const view = encode(JSON.stringify(query['view']));
+      searchParamsDict['view'] = view;
```

Comment:
Check all fields if they are the default values or have the value properties. And only to store the fields with values to the URL parameters.